frontend/packages/helm-plugin/src/utils/helm-detection-utils.ts

```diff
@@ -1,30 +1,48 @@
 import { SetFeatureFlag } from '@console/dynamic-plugin-sdk';
 import { fetchK8s } from '@console/internal/graphql/client';
 import { K8sResourceKind, ListKind } from '@console/internal/module/k8s';
-import { FLAG_OPENSHIFT_HELM } from '../const';
-import { HelmChartRepositoryModel } from '../models';
+import { useFlag } from '@console/shared/src/hooks/flag';
+import { FLAG_OPENSHIFT_HELM, FLAG_OPENSHIFT_CLUSTER_HELM } from '../const';
+import { HelmChartRepositoryModel, ProjectHelmChartRepositoryModel } from '../models';
 
-const checkDisabledHelmCharts = (helmChartRepositories: K8sResourceKind[]): boolean => {
+export const checkDisabledHelmCharts = (helmChartRepositories: K8sResourceKind[]): boolean => {
   let isDisabled = true;
-  helmChartRepositories.forEach((hcr) => {
+  helmChartRepositories?.forEach((hcr) => {
     isDisabled = isDisabled && (hcr.spec.disabled || false);
   });
   return isDisabled;
 };
 
-export const detectHelmChartRepositories = async (setFeatureFlag: SetFeatureFlag) => {
+export const detectHelmChartRepositories = async (setFeatureFlag: SetFeatureFlag, ns?: string) => {
   let id = null;
-  const fetchHelmChartRepositories = () =>
-    fetchK8s<ListKind<K8sResourceKind>>(HelmChartRepositoryModel)
+  const flag = ns ? FLAG_OPENSHIFT_HELM : FLAG_OPENSHIFT_CLUSTER_HELM;
+  const fetchHelmChartRepositories = () => {
+    const helmChartRepos: Promise<ListKind<K8sResourceKind>> = ns
+      ? fetchK8s<ListKind<K8sResourceKind>>(ProjectHelmChartRepositoryModel, null, ns)
+      : fetchK8s<ListKind<K8sResourceKind>>(HelmChartRepositoryModel);
+    helmChartRepos
       .then((list) => {
-        setFeatureFlag(FLAG_OPENSHIFT_HELM, !checkDisabledHelmCharts(list?.items));
+        setFeatureFlag(flag, !checkDisabledHelmCharts(list?.items));
+        if (ns) {
+          clearInterval(id);
+        }
```

Comment:
This doesn't seem right. We must not stop the detection if the user is in a context of a namespace, otherwise when the only ProjectHelmChartRepository CR is deleted in the namespace(with no cluster scoped HCRs), the Helm feature is not disabled.

Also, you would need to clear the interval when the active namespace changes, otherwise the timer will continue to run the detection for the old namespace along with the new one.